gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixTaskStateTracker.java

```diff
@@ -61,7 +61,9 @@ public GobblinHelixTaskStateTracker(Properties properties) {
   @Override
   public void registerNewTask(Task task) {
     try {
-      this.scheduledReporters.put(task.getTaskId(), scheduleTaskMetricsUpdater(new TaskMetricsUpdater(task), task));
+      if (GobblinMetrics.isEnabled(task.getTaskState().getWorkunit())) {
+        this.scheduledReporters.put(task.getTaskId(), scheduleTaskMetricsUpdater(new TaskMetricsUpdater(task), task));
```

Comment:
Using the global "metrics.enabled" key to turn off task metrics reporting seems too heavy. This global config will turn off other reporting as well, e.g. GTEs, and other metrics.

Reply:
I think a better approach might be to not create one metrics updater per task, and create a single updater instead, which iterates over all the tasks and updates records/bytes written.

Reply:
the additional thing being placed within the condition of `metrics.enabled` to be true, a.k.a `TaskMetricsUpdater` seems to be pretty metrics related and lightweight as it calls `updateRecordMetrics` and `updateByteMetrics`. Both of them are included if instrumented writer is being used, so I believe it is fair to gate these behavior with this config. 

If it is reversed, i.e. the `metrics.enabled` is controlling lightweight metrics-reporting behavior while we are trying to let it control the emission of GTE then I will be concerned as well. Let me know if it makes sense.

Reply:
Agreed. TaskMetrics collection seems un-necessary if metrics.enabled = false.